File: Examples/SpreadSheetsViewer/SimplePersonalBudget.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Reflection;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.SpreadSheetsViewer
{
    public class SimplePersonalBudget : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            //Load template file
            var fileStream = this.GetResourceStream("xlsx\\Simple personal budget.xlsx");
            workbook.Open(fileStream);
        }

        public override bool HasTemplate
        {
            get
            {
                return true;
            }
        }
        
        public override bool IsViewReadOnly
        {
            get
            {
                return false;
            }
        }

        public override bool ShowCode
        {
            get
            {
                return false;
            }
        }

        public override string TemplateName
        {
            get
            {
                return "Simple personal budget.xlsx";
            }
        }


        public override bool CanDownloadZip => false;
        public override string[] UsedResources
        {
            get
            {
                return new string[] { "xlsx\\Simple personal budget.xlsx" };
            }
        }
    }
}

File: Examples.VB/SpreadSheetsViewer/SimplePersonalBudget.vb
Namespace SpreadSheetsViewer
    Public Class SimplePersonalBudget
        Inherits ExampleBase

        Public Overrides Sub Execute(workbook As Workbook)
            'Load template file
            Dim fileStream = GetResourceStream("Simple personal budget.xlsx")
            workbook.Open(fileStream)
        End Sub

        Public Overrides ReadOnly Property HasTemplate As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property IsViewReadOnly As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property ShowCode As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property TemplateName As String
            Get
                Return "Simple personal budget.xlsx"
            End Get
        End Property


        Public Overrides ReadOnly Property CanDownloadZip As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property UsedResources As String()
            Get
                Return New String() {"xlsx\Simple personal budget.xlsx"}
            End Get
        End Property
    End Class
End Namespace
